sonic-xcvrd/xcvrd/xcvrd_utilities/xcvr_table_helper.py

```diff
@@ -238,3 +239,52 @@ def is_npu_si_settings_update_required(self, lport, port_mapping):
 
         # If npu_si_settings_sync_val is None, it can also mean that the key is not present in the table
         return npu_si_settings_sync_val is None or npu_si_settings_sync_val == NPU_SI_SETTINGS_DEFAULT_VALUE
+
+    def get_gearbox_line_lanes_dict(self):
+        """
+        Retrieves the gearbox line lanes dictionary from APPL_DB
+
+        This method scans all ASICs for gearbox interface configurations and extracts
+        the line_lanes count for each logical port. The line_lanes represent the
+        number of lanes on the line side (towards the optical module) which is the
+        correct count to use for CMIS host lane configuration.
+
+        Returns:
+            dict: A dictionary where:
+                - key (str): logical port name (e.g., "Ethernet0")
+                - value (int): number of line lanes for that port
+
+        Example:
+            {"Ethernet0": 2, "Ethernet200": 4}
+
+        Note:
+            - Returns empty dict if no gearbox configuration is found
+            - Silently skips invalid or malformed entries
+            - Only processes keys that start with "interface:"
+        """
+        gearbox_line_lanes_dict = {}
+        try:
+            for asic_id in self.appl_db:
+                appl_db = self.appl_db[asic_id]
+                gearbox_table = swsscommon.Table(appl_db, "_GEARBOX_TABLE")
```

Comment:
@aditya-nexthop can you make this namespace aware for platforms that may have multiple ASICs?